packages/workshop-app/app/components/touched-files.tsx

```diff
@@ -1,10 +1,99 @@
+import * as React from 'react'
+import { Await, useLoaderData } from '@remix-run/react'
 import * as Popover from '@radix-ui/react-popover'
+import { type loader } from '~/routes/_app+/_exercises+/$exerciseNumber_.$stepNumber.$type'
+import { LaunchEditor } from '~/routes/launch-editor'
 import Icon from './icons'
+import { SetAppToPlayground } from '~/routes/set-playground'
+
+function TouchedFiles() {
+	const data = useLoaderData<typeof loader>()
+	const fileListRef = React.useRef<{ name: string; children: JSX.Element }>()
+
+	const [open, setOpen] = React.useState(false)
+	const contentRef = React.useRef<HTMLDivElement>(null)
+
+	function handleLaunchUpdate() {
+		setOpen(false)
+	}
+
+	function getFileList() {
```

Comment:
This function has me pretty confused. Why are we doing this?

Reply:
I was probably more confused than you when i wrote this code.

I will fix this